File: src/sht30-test.py
import sys
import sht30
from time import sleep

from typing import *

TH_NUM_FIELDS: int = 3
DEBUG: int = 1

def main() -> None:
  stderr_fileno: int = sys.stderr
  sys.stderr = open("errors.log", "w")

  fd: int = sht30.initialize("/dev/i2c-1")
  cnt: int = 0

  if (fd >= 0):
    temp: list[int] = [0 for i in range(2)]
    temp_hum_arr: list[float] = [0.0 for i in range(TH_NUM_FIELDS)]
    while cnt < 50:
      if (sht30.processData(fd, temp, temp_hum_arr) >= 0):
        if (DEBUG):
          print("\nTemperature:", temp[0])
          print("Temperature (C):", temp_hum_arr[0])
          print("Temperature (F):", temp_hum_arr[1])
          print("Humidity:", temp_hum_arr[2])
      sleep(2)
      cnt += 1
  sys.stderr.close()
  sys.stderr = stderr_fileno

main()

File: Old/sht30-test.cpp
#include "sht30.hpp"

ofstream errorLog("errors.log");

int main() 
{
		//redirect cerr output to error log file
		cerr.rdbuf(errorLog.rdbuf());
	
    int32_t file = initialize();
    if (file >= 0) {
        uint16_t temp = 0;
        //
        float temp_hum_arr[TH_NUM_FIELDS] = {0.0};
        uint16_t count = 0;
        do {
	    if (processData(file, &temp, temp_hum_arr) >= 0) {
            // set data for LoRa transmission
						if (DEBUG) {
								cout << endl << "Temperature: " << temp << endl;
								cout << "Temperature (C): " << (double)temp_hum_arr[0] << endl;
								cout << "Temperature (F): " << (double)temp_hum_arr[1] << endl;
								cout << "Humidity: " << (double)temp_hum_arr[2] << endl;
						}
        }
	sleep(2);
	} while (++count < 10);
	close(file);
    }
    errorLog.close();
    return 0;
}